src/OpenTelemetry.Exporter.OneCollector/Internal/ExtensionFieldInformation.cs

```diff
@@ -0,0 +1,29 @@
+// <copyright file="ExtensionFieldInformation.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Text.Json;
+
+namespace OpenTelemetry.Exporter.OneCollector;
+
+internal sealed class ExtensionFieldInformation
+{
+    public string? ExtensionName;
+    public JsonEncodedText EncodedExtensionName;
+    public string? FieldName;
+    public JsonEncodedText EncodedFieldName;
+
+    public bool IsValid => this.ExtensionName != null;
```

Comment:
is `!=` make it valid or does it need to be non-empty?

Reply:
This class is sort of the final results that we keep in the cache. `null`, empty, or whitespace gets folded into the `null`/`!IsValid` case in [this logic](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/6a5d89b60c910b0164935a7ace4f2bf6b4b9750a/src/OpenTelemetry.Exporter.OneCollector/Internal/ExtensionFieldInformationManager.cs#L220-L260).